lib/src/shortest_path.dart

```diff
@@ -74,15 +74,14 @@ Map<T, List<T>> shortestPaths<T>(
 Map<T, List<T>> _shortestPaths<T>(
   T start,
   Iterable<T> Function(T) edges, {
-  T target,
-  bool Function(T, T) equals,
-  int Function(T) hashCode,
+  T? target,
+  bool Function(T?, T?)? equals,
+  int Function(T)? hashCode,
 }) {
   assert(start != null, '`start` cannot be null');
-  assert(edges != null, '`edges` cannot be null');
 
   final distances = HashMap<T, List<T>>(equals: equals, hashCode: hashCode);
-  distances[start] = List(0);
+  distances[start] = const [];
 
   equals ??= _defaultEquals;
   if (equals(start, target)) {
```

Comment:
This might allow you to change equals such that it doesn't have to accept a nullable T?
```suggestion
  if (target != null && equals(start, target)) {
```

Reply:
sure